src/test/java/com/puppycrawl/tools/checkstyle/PackageNamesLoaderTest.java

```diff
@@ -85,25 +78,20 @@ public void testDefault()
     @Test
     @SuppressWarnings("unchecked")
     public void testPackagesFile() throws Exception {
```

Comment:
Since you are rewriting this with new mocking, please add a javadoc explaining why we need mocking. See https://github.com/checkstyle/checkstyle/issues/4970 .

Reply:
done